extensions/wikia/MyHome/SpecialWikiActivity.class.php

```diff
@@ -9,150 +9,147 @@ class SpecialWikiActivity extends UnlistedSpecialPage {
 	private $feedSelected;
 
 	function __construct() {
-		parent::__construct('WikiActivity', '' /* no restriction */, true /* listed */);
+		parent::__construct( 'WikiActivity', '' /* no restriction */, true /* listed */ );
 	}
 
-	function execute($par) {
-		wfProfileIn(__METHOD__);
+	function execute( $par ) {
+		wfProfileIn( __METHOD__ );
 		global $wgOut, $wgUser, $wgBlankImgUrl, $wgEditPageFrameOptions;
 
 		$wgEditPageFrameOptions = "SAMEORIGIN";
 		$this->setHeaders();
 
 		// not available for skins different than Oasis
-		if (!F::app()->checkSkin('oasis')) {
+		if ( !F::app()->checkSkin( 'oasis' ) ) {
 			$wgOut->addWikiMsg( 'myhome-switch-to-monaco' );
-			wfProfileOut(__METHOD__);
+			wfProfileOut( __METHOD__ );
+
 			return;
 		}
 
 		// choose default view (RT #68074)
-		if ($wgUser->isLoggedIn()) {
+		if ( $wgUser->isLoggedIn() ) {
 			$this->defaultView = MyHome::getDefaultView();
-			if ($par == '') {
+			if ( $par == '' ) {
 				$par = $this->defaultView;
 			}
-		}
-		else {
+		} else {
 			$this->defaultView = false;
 		}
 
 		// watchlist feed
-		if($par == 'watchlist') {
+		if ( $par == 'watchlist' ) {
 			$this->classWatchlist = "selected";
 
 			// not available for anons
-			if($wgUser->isAnon()) {
-				if (get_class(RequestContext::getMain()->getSkin()) == 'SkinOasis') {
-					$wgOut->wrapWikiMsg( '<div class="latest-activity-watchlist-login" >$1</div>', array('oasis-activity-watchlist-login', wfGetReturntoParam()) );
-				}
-				else {
-					$wgOut->wrapWikiMsg( '<div id="myhome-log-in">$1</div>', array('myhome-log-in', wfGetReturntoParam()) );
+			if ( $wgUser->isAnon() ) {
+				if ( get_class( RequestContext::getMain()->getSkin() ) == 'SkinOasis' ) {
+					$wgOut->wrapWikiMsg( '<div class="latest-activity-watchlist-login" >$1</div>', [ 'oasis-activity-watchlist-login', wfGetReturntoParam() ] );
+				} else {
+					$wgOut->wrapWikiMsg( '<div id="myhome-log-in">$1</div>', [ 'myhome-log-in', wfGetReturntoParam() ] );
 				}
 
 				//oasis-activity-watchlist-login
 				// RT #23970
-				$wgOut->addInlineScript(<<<JS
+				$wgOut->addInlineScript( <<<JS
 $(function() {
 	$('#myhome-log-in').find('a').click(function(ev) {
 		openLogin(ev);
 	});
 });
 JS
 				);
-				wfProfileOut(__METHOD__);
+				wfProfileOut( __METHOD__ );
+
 				return;
-			}
-			else {
+			} else {
 				$this->feedSelected = 'watchlist';
 				$feedProxy = new WatchlistFeedAPIProxy();
-				$feedRenderer = new WatchlistFeedRenderer();
+				new WatchlistFeedRenderer();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either remove this useless object instantiation of class "WatchlistFeedRenderer" or use it [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1848)